src/transformStateWithClones.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  return actions.reduce((stateHistory, action) => {
+    const prevState = stateHistory[stateHistory.length - 1] || initialState;
+    let newState;
```

Comment:
**Well done on completing the task! 👍**  In the checklist, it is mentioned that clear names should be used for object copies. For example, instead of just using _newState_, you can use _stateCopy_ to better describe that it is a copy of the state.